File: web-app/online-warehouse/src/app/company/dto/create-company.dto.ts
import {CompanySizeEnum} from "./enum/company-size.enum";
import {CreateUserDto} from "../../user/dto/create-user.dto";

export class CreateCompanyDto {
  constructor(
    public name: string,
    public sizeType: CompanySizeEnum,
    public createUserDto: CreateUserDto
  ) {
  }
}

File: core/src/main/java/by/itechart/company/dto/CreateCompanyDto.java
package by.itechart.company.dto;

import by.itechart.common.dto.CreateUserDto;
import by.itechart.company.enums.CompanySize;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
public class CreateCompanyDto {
    @NotEmpty
    private String name;
    @NotNull
    private CompanySize sizeType;
    @NotNull
    private CreateUserDto createUserDto;
    private String logo;
}
